File: Data Structures & Algorithms/jump-game/submission-2.cpp
class Solution {
public:
    bool dfs(int i,vector<int>& nums){
        if(i== nums.size() -1 ) return true ;
        else{
            if(nums[i]==0) return false;
            else{
                for(int j=1 ;j<= nums[i];j++){
                    if(i+j< nums.size() && dfs(i+j,nums)) return true;
                }
            }
        }
        return false;
    }
    bool canJump(vector<int>& nums) {
        return dfs(0,nums);
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        d={}
        n=len(nums)
        def dfs(index):
            if index <0:
                return
            val=nums[index]
            if val in d:
                i=index+1
                while nums[i]!=val:
                    if val < nums[i]:
                        d[val]=max(d[nums[i]]+1,d[val])
                    i+=1
                dfs(index-1)
                return
            count=0
            for i in range(index+1,n):
                if val < nums[i]:
                    count=max(count,d[nums[i]])
            d[val]=count+1
            dfs(index-1)
        dfs(n-1)
        return max([x for x in d.values()])